olive/evaluator/olive_evaluator.py

```diff
@@ -312,6 +314,49 @@ def format_input(input_data, io_config):
             if k in input_names
         }
 
+    @staticmethod
+    def prepare_io_bindings(session, input_data, device, device_id: int = 0, shared_kv_buffer: bool = False):
+        """Convert input from numpy array to OrtValue.
+
+        session: ONNXRuntime session
+        input_data: dict of input data, value is numpy array
+        device: olive device
+        device_id: 0 by default. TODO(trajep): support user to specified device id
+        shared_kv_buffer: whether to share the key/value buffer across multiple runs, it is False by default,
+            and only used when we observe kv cache and fp16 is used.
+            TODO(trajep): how shared_kv_buffer works with generation task
+        """
+        from onnxruntime import OrtValue
+
+        use_fp16 = any(v.dtype == np.float16 for v in input_data.values())
+        io_bind_op = session.io_binding()
+        io_bind_device = "cuda" if device == "gpu" else "cpu"
+
+        if shared_kv_buffer:
+            OnnxEvaluator.KV_CACHE_ORTVALUES = OnnxEvaluator.KV_CACHE_ORTVALUES or {}
+
+        for k, v in input_data.items():
+            if shared_kv_buffer and use_fp16 and ("cache" in k or "past_key_values" in k):
```

Comment:
Could you add a comment about `"cache"` being from the microsoft version model (llama) and `past_key_values` from the hf version of the model?